lib/linux.js

```diff
@@ -123,15 +113,39 @@ function setAvailableApps() {
 	});
 }
 
+function isCinnamon() {
+	const args = ['writable', 'org.cinnamon.desktop.background', 'picture-uri'];
+	// output = childProcess.execFile('gsettings', args, ());
+	return cp.execFile('gsettings', args)
+		.then(out => (out.toString().trim() === 'true'))
+		.catch(err => {
+			err.toString(); // do nothing, this error is expected
+		});
+}
+
 exports.get = function get() {
 	if (!availableApps) {
 		return setAvailableApps().then(get);
 	}
 
 	const app = availableApps.find(app => app.get);
 
-	if (/nitrogen/.test(app.cmd)) {
-		app.cmd = '/bin/sed';
+	if (app.cmd === 'nitrogen') {
+		const homeDir = require('os').homedir();
+		const configFile = path.join(homeDir, '.config/nitrogen/bg-saved.cfg');
+		return fs.readFile(configFile, 'utf8').then(config => {
+			config = config.toString().trim().split('\n');
+			const wallpaper = config.find(line => /file=/.test(line));
+			return wallpaper.replace('file=', '');
+		});
```

Comment:
Could simply be:

```js
return fs.readFile(configFile, 'utf8').then(config =>
	config.trim().split('\n').find(line => /file=/.test(line)).replace('file=', '')
);
```

I don't know what the format is like, but I assume you want to remove `file=` from the start of the line, so it should be `/^file=/`. Same with replace.